aiida/tools/dbimporters/plugins/matp.py

```diff
@@ -0,0 +1,217 @@
+        try:
+            query = kwargs['query']
+        except AttributeError:
+            raise AttributeError('Make sure the supplied dictionary has `query` as a key. This '
+                                 'should contain a dictionary with the right query needed.')
+        try:
+            collection = kwargs['collection']
+        except AttributeError:
+            raise AttributeError('Make sure the supplied dictionary has `collection` as a key.')
+
+        if not isinstance(query, dict):
+            raise TypeError('The query argument should be a dictionary')
+
+        if collection is None:
+            collection = self._collection
+
+        if collection == 'structure':
+            results = []
+            collection_list = ['structure', 'material_id', 'cif']
+            for entry in self.find(query, collection_list):
+                results.append(entry)
+            search_results = MatProjSearchResults(results, return_class=MatProjCifEntry)
+        else:
+            raise ValueError('Unsupported collection: {}'.format(collection))
+
+        return search_results
+
+    def find(self, query, collection):
+        """
+        Query the database with a given dictionary of query parameters
+
+        :param query: a dictionary with the query parameters
+        """
+        for entry in self._mpr.query(criteria=query, properties=collection):
+            yield entry
+
+
+class MatProjEntry(DbEntry):
+    """
+    Represents an Materials Project database entry
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Set the class license from the source dictionary
+        """
+        lic = kwargs.pop('license', None)
+
+        if lic is not None:
+            self._license = lic
+
+        super(MatProjEntry, self).__init__(**kwargs)
+
+
+class MatProjCifEntry(CifEntry, MatProjEntry):  # pylint: disable=abstract-method
+    """
+    An extension of the MatProjEntry class with the CifEntry class, which will treat
+    the contents property through the URI as a cif file
+    """
+
+    def __init__(self, url, **kwargs):
+        """
+        The DbSearchResults base class instantiates a new DbEntry by explicitly passing the url
+        of the entry as an argument. In this case it is the same as the 'uri' value that is
+        already contained in the source dictionary so we just copy it
+        """
+        cif = kwargs.pop('cif', None)
+        kwargs['uri'] = url
+        super(MatProjCifEntry, self).__init__(**kwargs)
+
+        if cif is not None:
+            self.cif = cif
+
+
+class MatProjSearchResults(DbSearchResults):  # pylint: disable=abstract-method
+    """
+    A collection of MatProjEntry query result entries.
+    """
+
+    _db_name = 'Materials Project'
+    _db_uri = 'https://materialsproject.org'
+    _material_base_url = 'https://materialsproject.org/materials/'
+    _license = 'Unknown'
+    _version = 'Pulled from the Materials Project databse at: ' + str(datetime.datetime.now())
```

Comment:
I suggest leaving only:
```Python
_version = str(datetime.datetime.now())
```
Otherwise there is a typo in word ``databse``.

Reply:
Agreed. Is is self explanatory from the db_uri etc. Fixed.